web/src/popup_banners.ts

```diff
@@ -24,8 +25,18 @@ export function open_connection_error_popup_banner(opts: {
     on_retry_callback: () => void;
     is_get_events_error?: boolean;
 }): void {
-    // If the banner is already open, don't open it again.
-    if ($("#popup_banners_wrapper").find(".connection-error-banner").length > 0) {
+    // If the banner is already open, don't open it again, and instead remove
+    // the loading indicator on the retry button, if it was being shown.
+    const $banner = $("#popup_banners_wrapper").find(".connection-error-banner");
+    if ($banner.length > 0) {
+        const $retry_connection_button = $banner.find(".retry-connection");
+        if ($retry_connection_button.find(".button-loading-indicator").length > 0) {
+            // Add some delay before hiding the loading indicator, to visually
+            // indicate that the retry sequence was executed again but failed.
+            setTimeout(() => {
+                buttons.hide_button_loading_indicator($retry_connection_button);
+            }, 1000);
```

Comment:
This is a bit of a quirky way to do things. I don't think it's worse than what we had in `main`, though.